### DISCUSSION TITLE
Silly question about Birdnet-Pi

### DISCUSSION THREAD

**Discussion by DaveBKC** [alexbelgium/hassio-addons#1462] [Q&A] [answered]
I have Birdnet-Pi running on a dedicated RPi-4. Will this integration pull data from that dedicated device, or does this integration replace that device?

I'm running HA within Proxmox on an Intel NUC, so I'm trying to figure out if I can use this add-on. Thanks!

**Comment by alexbelgium** [comment, marked as answer]
Hi, it replaces the dedicated device. So indeed it would be a self contained birdnet-pi within your proxmox, with the benefits of avoiding a supplementary device, automated backups and such

Pulling data is also possible whatever the dévie, but it's a different topic, here is some ressources : https://community.home-assistant.io/t/birdnet-discussion/742670/6?u=alexbelgium

**Reply by DaveBKC** [reply]
Thanks for the fast response, Alex. My mics are mounted outside the house and the Pi mounted far away from my Proxmox, so I don't have the right use case for this right now. I routinely trash things though, so in a month things will likely change. :-)

**Reply by alexbelgium** [reply]
Hi my mic are at the end of the garden also ; what I did is convert my external pi in a rtsp server that sends the USB mic signal through rtsp to my HA birdnet-pi addon

But indeed if you have a working system I wouldn't bother either in your place to switch ;)  and a rpi4 has well enough power for the b-pi!

**Reply by UlrichThiess** [reply]
I search for a solution to use a mic as RTSP Server on an Raspberry. vlc or ffmpeg?

**Reply by alexbelgium** [reply]
Hi, here is the guide I followed to do exactly that : https://github.com/mcguirepr89/BirdNET-Pi/discussions/1006#discussioncomment-6747450

And my own customizations of those instructions : 
# On your desktop
Download imager
Install raspbian lite 64

# With ssh
sudo apt-get update
sudo apt-get install -y micro ffmpeg lsof
sudo -s cd /root && wget -c https://github.com/bluenviron/mediamtx/releases/download/v1.8.3/mediamtx_v1.8.3_linux_arm64v8.tar.gz -O - | sudo tar -xz

# List audio devices
arecord -l

# Check audio device parameters
arecord -D hw:1,0 --dump-hw-params
arecord -D hw:2 --dump-hw-params

# Add startup script
sudo nano startmic.sh
```
#!/bin/bash
echo "Starting birdmic"
# Disable gigabit ethernet
sudo ethtool -s eth0 speed 100 duplex full autoneg on
# Start rtsp server
./mediamtx & true
# Create rtsp feed
sleep 5
ffmpeg -nostdin -f alsa -acodec pcm_s16le -ac 2 -ar 48000 -i hw:1,0 -f rtsp -acodec pcm_s16be rtsp://localhost:8554/birdmic -rtsp_transport tcp || true & true

# Set microphone volume
sleep 5
amixer -c 1 sset Mic 71%
```

**Reply by UlrichThiess** [reply]
:-(
Administrator@BirdNET-Pi4:~ $ arecord -l
**** List of CAPTURE Hardware Devices ****
card 1: Device [USB PnP Sound Device], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
Administrator@BirdNET-Pi4:~ $ sudo ./startmic.sh 
Starting birdmic
2024/08/09 17:17:30 INF MediaMTX v1.8.3
2024/08/09 17:17:30 INF configuration loaded from /home/Administrator/mediamtx.yml
2024/08/09 17:17:30 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2024/08/09 17:17:30 INF [RTMP] listener opened on :1935
2024/08/09 17:17:30 INF [HLS] listener opened on :8888
2024/08/09 17:17:30 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
2024/08/09 17:17:30 INF [SRT] listener opened on :8890 (UDP)
ffmpeg version 5.1.5-0+rpt1+deb12u1 Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 12 (Debian 12.2.0-14)
  configuration: --prefix=/usr --extra-version=0+rpt1+deb12u1 --toolchain=hardened --incdir=/usr/include/aarch64-linux-gnu --enable-gpl --disable-stripping --disable-mmal --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sand --enable-sdl2 --disable-sndio --enable-libjxl --enable-neon --enable-v4l2-request --enable-libudev --enable-epoxy --libdir=/usr/lib/aarch64-linux-gnu --arch=arm64 --enable-pocketsphinx --enable-librsvg --enable-libdc1394 --enable-libdrm --enable-vout-drm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
Trailing option(s) found in the command: may be ignored.
[alsa @ 0x55753d0670] cannot set channel count to 2 (Invalid argument)
hw:1,0: Input/output error
Simple mixer control 'Mic',0
  Capabilities: cvolume cvolume-joined cswitch cswitch-joined
  Capture channels: Mono
  Limits: Capture 0 - 16
  Mono: Capture 11 [69%] [16.36dB] [on]

**Reply by alexbelgium** [reply]
Looks like your mic doesn't support stereo. You could try to tweak your rtsp command such as : `ffmpeg -nostdin -f alsa -acodec pcm_s16le -ac 1 -ar 48000 -i hw:1,0 -f rtsp -acodec pcm_s16be rtsp://localhost:8554/birdmic -rtsp_transport tcp || true & true`
